### ISSUE TITLE
"Govee Hub -Write Handler Slow to respond" - issue cont'd from #212

### ISSUE THREAD

**Issue by CadLiara** [bwp91/homebridge-govee#291]
What issue do you have? Please be as thorough and explicit as possible.

Any time I use voice commands to change power state, color, brightness on a room/more than 2-3 accessories or if any of those changes are effected by a scene/automation or triggered by something like a motion sensor, the lights are slow to respond (each change happens about 8-10+ seconds after the previous) and I get a Siri reply that the "lights/accessories were slow to respond" accompanied with an error in the logs (attached.)

I reference issue #212 as it is similar in nature to when I reported that. I had been running a 4. series beta (that you had recommended and things were ok, but when I upgraded to 5.2 this began (again.)

Troubleshooting steps taken:
- I have tried every currently available release and Beta version of the plugin.
- I have restarted the plugin, HOOBS, and the Govee app
- I have disabled all security settings (temporarily) on my Eero Pro 6 router. 
- I have reset and reflashed the HOOBS box. 
I have done MANY more steps (can go into detail) finally deleting my entire Homekit home, resetting every device, flashing the HOOBS box and doing everything to start from scratch with the re-setup starting with a new "home" in homekit and ONLY installing my apple TV and homepod minis, then HOOBS and then solely the Govee plugin (5.2 release version) just to see eliminate EVERY other possible plugin, device, app, etc and the problem persists. So I am at my wits end. 

I am not sure if this is a HOOBS issue, a Govee plugin issue, the number of Govee devices I am running issue, a homepod mini issue, an Eero issue, or some combination therein, but I don't seem to have any problems with any of my other Hubs/Devices (including the Switchbot plugin that is the only other plugin I run on HOOBS) (both plugins run under their own sub-bridges on HOOBS with Ciao identifier)

Details of your setup.
Do you use (1) Homebridge UI-X (2) Homebridge CLI or (3) HOOBS?
HOOBS in a box

Which version of Homebridge/HOOBS do you have?
4.2.8

Which platform do you run Homebridge/HOOBS on (e.g. Raspberry Pi/Windows/HOOBS Box)? Please also mention your version of Node.js/NPM if known.
HOOBS in a box
NODE: 16.16.0
HomeBridge Version 1.5.0

Which version of this plugin (homebridge-govee) do you have? Has the issue started since upgrading from a previous version?
5.2

Which Govee devices do you have that are causing issues? Please include product models if applicable.
It doesn't seem to matter as I have over a dozen different devices (models) and it seems to randomly effect almost all of them at different times/instances. Whether it is WiFi light strips, Neon, Glide, etc
[Govee plugin log 8-13-2022.txt](https://github.com/bwp91/homebridge-govee/files/9332537/Govee.plugin.log.8-13-2022.txt)
.

**Comment by CadLiara** [comment]
Bumping this due to being notified as it is being closed due to inactivity. The issue has worsened considerably and my govee devices are practically unusable at this point. Every request sent (either by voice through a homepod mini or via Siri on iphone/iwatch.) or any automation (like a motion sensor trigger) ends up taking FOREVER. 
After a request, it takes about 20-30 seconds for a light  to come on, and when multiple lights are present in the same automation/request, they come on in sequence with about 20 seconds to a minute in between each. 

This is compounded by nothing happening in parallel, so if an automation is triggered, I have to wait until everything in that automation happens before the next request is even processed. 

It has never been this bad. 

I did post in the HOOBS group on FB as one thing I did notice (as I tried updating from 5.2 to the 5.2.1beta 11) is that I see two instances of the Govee bridge that I can update to. When I select one, both become highlighted. 

I am guessing this has something to do with it. 
Per some advice on FB Hoobs group, I increased refresh interval to 90 and control interval to 10000 and changed the govee login pw in both the app and the plugin. 
No change. 

In any case, if you can let me know what you think and what I can do.

Thanks

**Comment by CadLiara** [comment]
Attaching most recent logs

(And please pardon the perhaps silly question, but I am noticing a lot of concurrent traffic for the same devices for both API and AWS. Not sure if that means anything.)
[Govee plugin log 9-8-2022.txt](https://github.com/bwp91/homebridge-govee/files/9531143/Govee.plugin.log.9-8-2022.txt)

**Comment by CadLiara** [comment]
I just reset HOOBS via factory reset and the first thing I installed was the govee plugin. Can you please advise on whether I am correctly suspicious or not regarding the error in install and then it seemingly duplicating the install?
Logs in next post.

**Comment by CadLiara** [comment]
9/8/2022, 7:02:36 PMReading package lists...
9/8/2022, 7:03:01 PMdeb [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_16.x bullseye main
9/8/2022, 7:03:01 PMdeb-src [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_16.x bullseye main
9/8/2022, 7:03:49 PMHit:1 https://deb.nodesource.com/node_16.x bullseye InRelease
9/8/2022, 7:03:49 PMHit:2 http://raspbian.raspberrypi.org/raspbian bullseye InRelease
9/8/2022, 7:03:49 PMHit:3 http://archive.raspberrypi.org/debian bullseye InRelease
9/8/2022, 7:03:49 PMHit:4 https://dl.yarnpkg.com/debian stable InRelease
9/8/2022, 7:03:49 PMGet:5 https://dl.hoobs.org/debian stable InRelease [3,032 B]
9/8/2022, 7:03:49 PMFetched 3,032 B in 4s (805 B/s)
9/8/2022, 7:03:49 PMReading package lists...
9/8/2022, 7:07:38 PMyarn add v1.22.19
9/8/2022, 7:07:38 PMinfo No lockfile found.
9/8/2022, 7:07:39 PM[1/4] Resolving packages...
9/8/2022, 7:07:44 PM[2/4] Fetching packages...
9/8/2022, 7:07:45 PM[3/4] Linking dependencies...
9/8/2022, 7:07:53 PM[4/4] Building fresh packages...
9/8/2022, 7:08:22 PMwarning Error running install script for optional dependency: "/var/lib/hoobs/goveebridge/node_modules/usb: Command failed.
9/8/2022, 7:08:22 PMExit code: 1
9/8/2022, 7:08:22 PMCommand: node-gyp-build
9/8/2022, 7:08:22 PMArguments:
9/8/2022, 7:08:22 PMDirectory: /var/lib/hoobs/goveebridge/node_modules/usb
9/8/2022, 7:08:22 PMOutput:
9/8/2022, 7:08:22 PMgyp info it worked if it ends with ok
9/8/2022, 7:08:22 PMgyp info using node-gyp@9.0.0
9/8/2022, 7:08:22 PMgyp info using node@16.17.0 | linux | arm
9/8/2022, 7:08:22 PMgyp info find Python using Python version 3.9.2 found at \"/usr/bin/python3\"
9/8/2022, 7:08:22 PMgyp info spawn /usr/bin/python3
9/8/2022, 7:08:22 PMgyp info spawn args [
9/8/2022, 7:08:22 PMgyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
9/8/2022, 7:08:22 PMgyp info spawn args   'binding.gyp',
9/8/2022, 7:08:22 PMgyp info spawn args   '-f',
9/8/2022, 7:08:22 PMgyp info spawn args   'make',
9/8/2022, 7:08:22 PMgyp info spawn args   '-I',
9/8/2022, 7:08:22 PMgyp info spawn args   '/var/lib/hoobs/goveebridge/node_modules/usb/build/config.gypi',
9/8/2022, 7:08:22 PMgyp info spawn args   '-I',
9/8/2022, 7:08:22 PMgyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
9/8/2022, 7:08:22 PMgyp info spawn args   '-I',
9/8/2022, 7:08:22 PMgyp info spawn args   '/root/.cache/node-gyp/16.17.0/include/node/common.gypi',
9/8/2022, 7:08:22 PMgyp info spawn args   '-Dlibrary=shared_library',
9/8/2022, 7:08:23 PMgyp info spawn args   '-Dvisibility=default',
9/8/2022, 7:08:23 PMgyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/16.17.0',
9/8/2022, 7:08:23 PMgyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/npm/node_modules/node-gyp',
9/8/2022, 7:08:23 PMgyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/16.17.0/<(target_arch)/node.lib',
9/8/2022, 7:08:23 PMgyp info spawn args   '-Dmodule_root_dir=/var/lib/hoobs/goveebridge/node_modules/usb',
9/8/2022, 7:08:23 PMgyp info spawn args   '-Dnode_engine=v8',
9/8/2022, 7:08:23 PMgyp info spawn args   '--depth=.',
9/8/2022, 7:08:23 PMgyp info spawn args   '--no-parallel',
9/8/2022, 7:08:23 PMgyp info spawn args   '--generator-output',
9/8/2022, 7:08:23 PMgyp info spawn args   'build',
9/8/2022, 7:08:23 PMgyp info spawn args   '-Goutput_dir=.'
9/8/2022, 7:08:23 PMgyp info spawn args ]
9/8/2022, 7:08:23 PMgyp info spawn make
9/8/2022, 7:08:23 PMgyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
9/8/2022, 7:08:23 PMmake: Entering directory '/var/lib/hoobs/goveebridge/node_modules/usb/build'
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/core.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/descriptor.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/hotplug.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/io.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/strerror.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/sync.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/os/poll_posix.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/os/threads_posix.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/os/linux_usbfs.o
9/8/2022, 7:08:23 PMCC(target) Release/obj.target/libusb/libusb/libusb/os/linux_udev.o
9/8/2022, 7:08:23 PM../libusb/libusb/os/linux_udev.c:40:10: fatal error: libudev.h: No such file or directory
9/8/2022, 7:08:23 PM40 | #include <libudev.h>
9/8/2022, 7:08:23 PM|          ^~~~~~~~~~~
9/8/2022, 7:08:23 PMcompilation terminated.
9/8/2022, 7:08:23 PMmake: *** [libusb.target.mk:156: Release/obj.target/libusb/libusb/libusb/os/linux_udev.o] Error 1
9/8/2022, 7:08:23 PMmake: Leaving directory '/var/lib/hoobs/goveebridge/node_modules/usb/build'
9/8/2022, 7:08:23 PMgyp ERR! build error
9/8/2022, 7:08:23 PMgyp ERR! stack Error: `make` failed with exit code: 2
9/8/2022, 7:08:23 PMgyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
9/8/2022, 7:08:23 PMgyp ERR! stack     at ChildProcess.emit (node:events:513:28)
9/8/2022, 7:08:23 PMgyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
9/8/2022, 7:08:23 PMgyp ERR! System Linux 5.15.56-v7+
9/8/2022, 7:08:23 PMgyp ERR! command \"/usr/bin/node\" \"/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"
9/8/2022, 7:08:23 PMgyp ERR! cwd /var/lib/hoobs/goveebridge/node_modules/usb
9/8/2022, 7:08:23 PMgyp ERR! node -v v16.17.0
9/8/2022, 7:08:23 PMgyp ERR! node-gyp -v v9.0.0
9/8/2022, 7:08:23 PMgyp ERR! not ok"
9/8/2022, 7:08:23 PMinfo This module is OPTIONAL, you can safely ignore this error
9/8/2022, 7:08:53 PMsuccess Saved lockfile.
9/8/2022, 7:08:53 PMsuccess Saved 93 new dependencies.
9/8/2022, 7:08:53 PMinfo Direct dependencies
9/8/2022, 7:08:53 PM└─ homebridge-govee@5.2.1-beta.11
9/8/2022, 7:08:53 PMinfo All dependencies
9/8/2022, 7:08:53 PM├─ @abandonware/noble@1.9.2-15
9/8/2022, 7:08:53 PM├─ @homebridge/plugin-ui-utils@0.0.19
9/8/2022, 7:08:53 PM├─ @mapbox/node-pre-gyp@1.0.10
9/8/2022, 7:08:53 PM├─ abbrev@1.1.1
9/8/2022, 7:08:53 PM├─ agent-base@6.0.2
9/8/2022, 7:08:53 PM├─ ansi-regex@5.0.1
9/8/2022, 7:08:53 PM├─ aproba@2.0.0
9/8/2022, 7:08:53 PM├─ are-we-there-yet@2.0.0
9/8/2022, 7:08:53 PM├─ async-limiter@1.0.1
9/8/2022, 7:08:53 PM├─ asynckit@0.4.0
9/8/2022, 7:08:53 PM├─ aws-iot-device-sdk@2.2.12
9/8/2022, 7:08:53 PM├─ axios@0.27.2
9/8/2022, 7:08:53 PM├─ balanced-match@1.0.2
9/8/2022, 7:08:53 PM├─ base64-js@1.5.1
9/8/2022, 7:08:53 PM├─ bl@4.1.0
9/8/2022, 7:08:53 PM├─ brace-expansion@1.1.11
9/8/2022, 7:08:53 PM├─ buffer-from@1.1.2
9/8/2022, 7:08:53 PM├─ buffer@5.7.1
9/8/2022, 7:08:53 PM├─ chownr@2.0.0
9/8/2022, 7:08:53 PM├─ color-support@1.1.3
9/8/2022, 7:08:53 PM├─ combined-stream@1.0.8
9/8/2022, 7:08:53 PM├─ commist@1.1.0
9/8/2022, 7:08:53 PM├─ concat-map@0.0.1
9/8/2022, 7:08:53 PM├─ concat-stream@2.0.0
9/8/2022, 7:08:53 PM├─ console-control-strings@1.1.0
9/8/2022, 7:08:53 PM├─ core-util-is@1.0.3
9/8/2022, 7:08:53 PM├─ crypto-js@4.0.0
9/8/2022, 7:08:53 PM├─ delayed-stream@1.0.0
9/8/2022, 7:08:53 PM├─ delegates@1.0.0
9/8/2022, 7:08:53 PM├─ detect-libc@2.0.1
9/8/2022, 7:08:53 PM├─ duplexify@3.7.1
9/8/2022, 7:08:53 PM├─ emoji-regex@8.0.0
9/8/2022, 7:08:53 PM├─ end-of-stream@1.4.4
9/8/2022, 7:08:53 PM├─ eventemitter3@4.0.7
9/8/2022, 7:08:53 PM├─ follow-redirects@1.15.1
9/8/2022, 7:08:53 PM├─ form-data@4.0.0
9/8/2022, 7:08:53 PM├─ fs-minipass@2.1.0
9/8/2022, 7:08:53 PM├─ fs.realpath@1.0.0
9/8/2022, 7:08:53 PM├─ gauge@3.0.2
9/8/2022, 7:08:53 PM├─ glob@7.2.3
9/8/2022, 7:08:53 PM├─ has-unicode@2.0.1
9/8/2022, 7:08:53 PM├─ help-me@3.0.0
9/8/2022, 7:08:53 PM├─ homebridge-govee@5.2.1-beta.11
9/8/2022, 7:08:53 PM├─ https-proxy-agent@5.0.1
9/8/2022, 7:08:53 PM├─ ieee754@1.2.1
9/8/2022, 7:08:53 PM├─ inflight@1.0.6
9/8/2022, 7:08:54 PM├─ inherits@2.0.4
9/8/2022, 7:08:54 PM├─ is-fullwidth-code-point@3.0.0
9/8/2022, 7:08:54 PM├─ isarray@1.0.0
9/8/2022, 7:08:54 PM├─ leven@2.1.0
9/8/2022, 7:08:54 PM├─ lru-cache@6.0.0
9/8/2022, 7:08:54 PM├─ make-dir@3.1.0
9/8/2022, 7:08:54 PM├─ mime-db@1.52.0
9/8/2022, 7:08:54 PM├─ mime-types@2.1.35
9/8/2022, 7:08:54 PM├─ minimatch@3.1.2
9/8/2022, 7:08:54 PM├─ minimist@1.2.6
9/8/2022, 7:08:54 PM├─ minizlib@2.1.2
9/8/2022, 7:08:54 PM├─ mkdirp@1.0.4
9/8/2022, 7:08:54 PM├─ mqtt-packet@6.10.0
9/8/2022, 7:08:54 PM├─ mqtt@4.2.8
9/8/2022, 7:08:54 PM├─ ms@2.1.2
9/8/2022, 7:08:54 PM├─ nan@2.16.0
9/8/2022, 7:08:54 PM├─ node-addon-api@3.2.1
9/8/2022, 7:08:54 PM├─ node-fetch@2.6.7
9/8/2022, 7:08:54 PM├─ node-persist@3.1.0
9/8/2022, 7:08:54 PM├─ nopt@5.0.0
9/8/2022, 7:08:54 PM├─ npmlog@5.0.1
9/8/2022, 7:08:54 PM├─ object-assign@4.1.1
9/8/2022, 7:08:54 PM├─ p-queue@7.3.0
9/8/2022, 7:08:54 PM├─ p-timeout@5.1.0
9/8/2022, 7:08:54 PM├─ path-is-absolute@1.0.1
9/8/2022, 7:08:54 PM├─ process-nextick-args@2.0.1
9/8/2022, 7:08:54 PM├─ pump@3.0.0
9/8/2022, 7:08:54 PM├─ readable-stream@3.6.0
9/8/2022, 7:08:54 PM├─ reinterval@1.1.0
9/8/2022, 7:08:54 PM├─ rimraf@3.0.2
9/8/2022, 7:08:54 PM├─ semver@7.3.7
9/8/2022, 7:08:54 PM├─ set-blocking@2.0.0
9/8/2022, 7:08:54 PM├─ signal-exit@3.0.7
9/8/2022, 7:08:54 PM├─ split2@3.2.2
9/8/2022, 7:08:54 PM├─ string_decoder@1.3.0
9/8/2022, 7:08:54 PM├─ string-width@4.2.3
9/8/2022, 7:08:54 PM├─ tar@6.1.11
9/8/2022, 7:08:54 PM├─ tr46@0.0.3
9/8/2022, 7:08:54 PM├─ typedarray@0.0.6
9/8/2022, 7:08:54 PM├─ ultron@1.1.1
9/8/2022, 7:08:54 PM├─ util-deprecate@1.0.2
9/8/2022, 7:08:54 PM├─ webidl-conversions@3.0.1
9/8/2022, 7:08:54 PM├─ websocket-stream@5.5.2
9/8/2022, 7:08:54 PM├─ whatwg-url@5.0.0
9/8/2022, 7:08:54 PM├─ wide-align@1.1.5
9/8/2022, 7:08:54 PM├─ ws@3.3.3
9/8/2022, 7:08:54 PM└─ xtend@4.0.2
9/8/2022, 7:08:54 PMDone in 75.25s.
9/8/2022, 7:08:56 PMGovee Bridge starting
9/8/2022, 7:10:01 PMGovee BridgeLoaded plugin 'homebridge-govee'
9/8/2022, 7:10:01 PMGovee BridgeLoaded plugin 'homebridge-govee'
9/8/2022, 7:10:01 PMGovee BridgeLoading 1 platforms...
9/8/2022, 7:10:01 PMGovee BridgeLoading 1 platforms...
9/8/2022, 7:10:01 PMGovee BridgeGoveeInitialising plugin v5.2.1-beta.11 | System linux | Node v16.17.0 | HB v1.5.0 | HAPNodeJS v0.10.2...
9/8/2022, 7:10:01 PMGovee BridgeGoveeInitialising plugin v5.2.1-beta.11 | System linux | Node v16.17.0 | HB v1.5.0 | HAPNodeJS v0.10.2...
9/8/2022, 7:10:01 PMGovee BridgeGoveePlugin initialised. Setting up accessories....
9/8/2022, 7:10:01 PMGovee BridgeGoveePlugin initialised. Setting up accessories....
9/8/2022, 7:10:01 PMGovee BridgeRegistering platform 'homebridge-govee.Govee'
9/8/2022, 7:10:01 PMGovee BridgeRegistering platform 'homebridge-govee.Govee'
9/8/2022, 7:10:01 PMGovee BridgeBridge is running on port 51826.
9/8/2022, 7:10:01 PMGovee BridgeBridge is running on port 51826.
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNINGDisabling HTTP client as username and/or password not provided.
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNINGDisabling HTTP client as username and/or password not provided.
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNINGDisabling API client as API key not provided.
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNINGDisabling API client as API key not provided.
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNINGDisabling BLE client as no enabled devices configured.
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNINGDisabling BLE client as no enabled devices configured.
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNING***** Disabling plugin. *****
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNING***** Disabling plugin. *****
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNING***** No devices found to initialise. *****
9/8/2022, 7:10:01 PMGovee BridgeGoveeWARNING***** No devices found to initialise. *****

**Comment by CadLiara** [comment]
Disregard. I was able to "fix" the issue. 

If it is helpful to others, here is what I did and what changed from previous installs of this plugin in case you run into similar. 

- Factory reset HOOBS.
- Created GOVEE bridge (same as before)
- Installed 5.2.1 beta 11 (bridge advertiser Bonjour. I was using Ciao before)
- ONLY input API token and NOT login credentials (In the past I had done both)
- Once I realized that AWS was needed for video mode, I used EVE to get all of my device ID's and in configuration I input all GOVEE devices and disabled AWS for all of them except the two devices I need to get an EVE scene in. (In the past I would only input in configure those two devices and none others, so both API and AWS was enabled for all but not BLE as I don't use BLE)

Not sure which of these steps/changes solved the issue, but now I am still getting errors in logs, but everything seems to be functioning correctly. The only time I get Siri telling me "devices are taking too long" is when I group command all devices in my office (I have 14 devices in that room)

So hope this helps anyone dealing with something similar. 

Needless to say, I did an immediate backup of this profile in case anything goes haywire and I need to go back.
